project/plugins/ssh.py

```diff
@@ -8,74 +8,97 @@
 logging.getLogger('paramiko').setLevel(logging.CRITICAL)
 
 
-def SSH_server(hostname, username, port, commands, password=None, pkey=None, marker=None, markers=None):
+def load_ssh_key(pkey_path, password=None):
+    try:
+        with open(pkey_path, 'r') as key_file:
+            key_data = key_file.read()
+
+        if "BEGIN RSA PRIVATE KEY" in key_data:
+            return paramiko.RSAKey.from_private_key_file(pkey_path, password=password)
+        elif "BEGIN OPENSSH PRIVATE KEY" in key_data:
+            return paramiko.Ed25519Key.from_private_key_file(pkey_path, password=password)
+        else:
+            raise ValueError("Unsupported key format")
+    except paramiko.PasswordRequiredException:
+        logging.error("SSH key is encrypted and requires a passphrase.")
+    except paramiko.SSHException as e:
+        logging.error(f"Error loading SSH key: {e}")
+    except Exception as e:
+        logging.error(f"Unexpected error: {e}")
+    return None
+
+
+def find_line_number(client, file_path, marker, password=None):
+    find_line_cmd = f"sed -n '/{marker}/=' {file_path}"
+    if password:
+        find_line_cmd = f"echo '{password}' | sudo -S {find_line_cmd}"
+
+    stdin, stdout, stderr = client.exec_command(find_line_cmd, get_pty=True)
+    output = stdout.read().decode("utf-8")
+    logging.debug(f"Output from find_line_cmd: {output}")
+
+    lines = re.findall(r'\d+', output)
+    if lines:
+        return int(lines[0])
+    else:
+        logging.error(f"No line number found for the marker: {marker}")
+        return None
+
+
+def update_env_vars(client, file_path, commands, markers, password=None):
```

Comment:
Not sure this is a good function name - it doesn't LOOK like we're updating environment variables...

Reply:
Kevin, In the past, we used to update the Docker Compose manifest directly to update the new access and secret keys. We now update the credentials in [the .env file](https://bitbucket.org/signiant/tool-configuration-files/src/b0bce12942655aaac96cdcf359ae37b692bfceb6/LOCK/config.yaml#lines-53,54), which is located at the same level as the Docker Compose file. The .env file allows us to manage environment variables more easily by referencing them directly in the Docker Compose manifest. We use markers to identify the line numbers for the sed command, which replaces the AWS access and secret keys in the .env file so I called the function update_env_vars. This is only applicable to our Nagios user.